site/src/pages/WorkspacesPage/WorkspacesButton.tsx

```diff
@@ -152,75 +62,186 @@ export function WorkspacesButton({
   }
 
   return (
-    <PopoverContainer
-      // Stopgap value until bug where string-based horizontal origin isn't
-      // being applied consistently can get figured out
-      originX={-115}
-      originY="bottom"
-      sx={{ display: "flex", flexFlow: "column nowrap" }}
-      anchorButton={
-        <Button startIcon={<AddIcon />} variant="contained">
-          {children}
-        </Button>
-      }
-    >
-      <SearchBox
-        value={searchTerm}
-        onValueChange={(newValue) => setSearchTerm(newValue)}
-        placeholder="Type/select a workspace template"
-        label="Template select for workspace"
-        sx={{ flexShrink: 0, columnGap: COLUMN_GAP }}
-      />
-
-      <OverflowY
-        maxHeight={380}
-        sx={{
-          display: "flex",
-          flexFlow: "column nowrap",
-          paddingY: 1,
+    <>
+      <Button
+        startIcon={<AddIcon />}
+        variant="contained"
+        ref={anchorRef}
+        onClick={() => {
+          setIsOpen(true);
         }}
       >
-        {templatesFetchStatus === "loading" ? (
-          <Loader size={14} />
-        ) : (
-          <>
-            {processed.map((template) => (
-              <WorkspaceResultsRow key={template.id} template={template} />
-            ))}
-
-            {emptyState}
-          </>
-        )}
-      </OverflowY>
-
-      <Link
-        component={RouterLink}
-        to="/templates"
-        sx={{
-          outline: "none",
-          "&:focus": {
-            backgroundColor: theme.palette.action.focus,
-          },
+        {children}
+      </Button>
+      <Popover
+        disablePortal
```

Comment:
This seems like a much nicer solution than my forced closing approach, but I did notice this from the MUI docs:
> The children is [Portal](https://mui.com/material-ui/react-portal/) to the body of the document to avoid rendering problems. You can disable this behavior with disablePortal.

I couldn't find anything about what these rendering problems are, though. Do you happen to know anything about this?

Reply:
I never had such problems but I would guess it can be related to style since the component would be rendered inside of the component scope and not outside of the #root element